File: Java/src/com/leetcode/Easy/KthMissingPositiveNumber.java
package com.leetcode.Easy;

// 1539. Kth Missing Positive Number
public class KthMissingPositiveNumber {
    public int findKthPositive(int[] arr, int k) {
        int left = 0, right = arr.length;
        while (left < right) {
            int mid = (left + right) / 2;
            if (arr[mid] - mid - 1 < k)
                left = mid + 1;
            else
                right = mid;
        }
        return left + k;
    }
}

File: JavaScript/Easy/KthMissingPositiveNumber.js
/**
 * 1539. Kth Missing Positive Number
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findKthPositive = function (arr, k) {
  let left = 0,
    right = arr.length;
  while (left < right) {
    let mid = Math.floor((left + right) / 2);
    if (arr[mid] - mid - 1 < k) left = mid + 1;
    else right = mid;
  }
  return left + k;
};
